src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3639,18 +3639,67 @@ void addToStringMethod(MethodSymbol printMethod)
                     refCustomModifiers: ImmutableArray<CustomModifier>.Empty,
                     explicitInterfaceImplementations: ImmutableArray<MethodSymbol>.Empty);
 
-                if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
+                var baseToStringMethod = getBaseToStringMethod();
+                bool isBaseToStringSealed = baseToStringMethod is { IsSealed: true };
+
+                if (!this.DeclaringCompilation.IsFeatureEnabled(MessageID.IDS_FeatureSealedToStringInRecord) && isBaseToStringSealed)
                 {
-                    var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
-                    members.Add(toStringMethod);
+                    if (baseToStringMethod?.ContainingModule != this.ContainingModule)
```

Comment:
>? [](start = 42, length = 1)

It feels like the conditional access is not necessary here.

Reply:
I know, but the compiler disagrees (CS8602 Dereference of a possible null reference). But you're right, I should use `!` instead of `?`, since I know it's not null.

Reply:
Done